modules/client/src/addressList/internal/client/methods.ts

```diff
@@ -344,17 +363,29 @@ export class ClientAddressListMethods extends ClientCore
       await this.ipfs.ensureOnline();
       return Promise.all(
         addressListProposals.map(
-          (
+          async (
             proposal: SubgraphAddressListProposalListItem,
           ): Promise<AddressListProposalListItem> => {
             // format in the metadata field
-            return this.ipfs
-              .fetchString(proposal.metadata)
-              .then((stringMetadata: string) => {
-                // TODO: Parse and validate schema
-                const metadata = JSON.parse(stringMetadata) as ProposalMetadata;
-                return toAddressListProposalListItem(proposal, metadata);
-              });
+            try {
+              const metadataCid = resolveIpfsCid(proposal.metadata);
+              const stringMetadata = await this.ipfs.fetchString(metadataCid);
+              const metadata = JSON.parse(stringMetadata) as ProposalMetadata;
+              return toAddressListProposalListItem(proposal, metadata);
+            } catch (err) {
+              if (err instanceof InvalidCidError) {
+                return Promise.resolve(
```

Comment:
If the inner function is `async` then we don't need to wrap everything with `Promise.resolve()`